--- song_data.h
+++ song_data.h
@@ -41,7 +41,7 @@
   //"Intro channel 0"
   0x40, 36,               // FX: SET VOLUME: volume = 36
   0x9E, 4, 5, 6, 7,       // FX: GOTO advanced: ch0 = 0x04 / ch1 = 0x05 / ch2 = 0x06 / ch3 = 0x07
-  0x41, -4,               // FX: SLIDE VOLUME ON: -4
+  0x41, static_cast<uint8_t>(-4),               // FX: SLIDE VOLUME ON: -4
   0x54, 2,                // FX: SET NOTE CUT: 2
   0x9D, 24,               // FX: SET TEMPO: tempo = 24
   0xFD, 6, 9,             // REPEAT: count = 6 + 1 / track = 9
@@ -52,7 +52,7 @@

   //"Intro channel 2"
   0x40, 64,               // FX: SET VOLUME: volume = 64
-  0x41, -6,               // FX: SLIDE VOLUME ON: -6
+  0x41, static_cast<uint8_t>(-6),               // FX: SLIDE VOLUME ON: -6
   0xFD, 1, 24,            // REPEAT: count = 1 + 1 / track = 24
   0xFC, 13,               // GOTO track 13
   0xFC, 14,               // GOTO track 14
@@ -78,7 +78,7 @@
   //"Track channel 4"
   0x40, 36,               // FX: SET VOLUME: volume = 36
   0x9E, 4, 5, 6, 7,       // FX: GOTO advanced: ch0 = 0x04 / ch1 = 0x05 / ch2 = 0x06 / ch3 = 0x07
-  0x41, -4,               // FX: SLIDE VOLUME ON: -4
+  0x41, static_cast<uint8_t>(-4),               // FX: SLIDE VOLUME ON: -4
   0x54, 2,                // FX: SET NOTE CUT: 2
   0x9D, 24,               // FX: SET TEMPO: tempo = 24
   0xFD, 5, 9,             // REPEAT: count = 5 + 1 / track = 9
@@ -89,7 +89,7 @@

   //"Track channel 6"
   0x40, 64,               // FX: SET VOLUME: volume = 64
-  0x41, -6,               // FX: SLIDE VOLUME ON: -6
+  0x41, static_cast<uint8_t>(-6),               // FX: SLIDE VOLUME ON: -6
   0xFC, 13,               // GOTO track 13
   0xFC, 14,               // GOTO track 14
   0xFC, 15,               // GOTO track 15
@@ -343,7 +343,7 @@

   //"Track channel 1"
   0x40, 48,               // FX: SET VOLUME: volume = 48
-  0x41, -1,               // FX: SLIDE VOLUME ON: -1
+  0x41, static_cast<uint8_t>(-1),               // FX: SLIDE VOLUME ON: -1
   0x50, 4, 1,             // FX: SET VIBRATO: 4 1
   0x54, 3,                // FX: SET NOTE CUT: 3
   0xFC, 5,                // GOTO track 5
@@ -351,7 +351,7 @@

   //"Track channel 2"
   0x40, 48,               // FX: SET VOLUME: volume = 48
-  0x41, -1,               // FX: SLIDE VOLUME ON: -1
+  0x41, static_cast<uint8_t>(-1),               // FX: SLIDE VOLUME ON: -1
   0x4E, 6, 1,             // FX: SET TREMOLO: 6 1
   0x40, 24,               // FX: SET VOLUME: volume = 24
   0xFC, 4,                // GOTO track 4
@@ -428,10 +428,10 @@
   //"Track channel 0"
   0x40, 48,               // FX: SET VOLUME: volume = 48
   0x9D, 25,               // FX: SET TEMPO: tempo = 25
-  0x41, -1,               // FX: SLIDE VOLUME ON: -1
+  0x41, static_cast<uint8_t>(-1),               // FX: SLIDE VOLUME ON: -1
   0xFC, 5,                // GOTO track 5
   0xFC, 9,                // GOTO track 9
-  0x41, -12,              // FX: SLIDE VOLUME ON: -12
+  0x41, static_cast<uint8_t>(-12),              // FX: SLIDE VOLUME ON: -12
   0xFC, 6,                // GOTO track 6
   0x9F,                   // FX: STOP CURRENT CHANNEL

@@ -449,7 +449,7 @@

   //"Track snare"
   0x40, 32,               // FX: SET VOLUME: volume = 32
-  0x41, -16,              // FX: VOLUME SLIDE ON: steps = -16
+  0x41, static_cast<uint8_t>(-16),              // FX: VOLUME SLIDE ON: steps = -16
   0x9F + 2,               // DELAY: ticks = 2
   0x43,                   // FX: VOLUME SLIDE OFF
   0xFE,                   // RETURN
