--- ARDU_AB.ino
+++ ARDU_AB.ino
@@ -64,11 +64,6 @@ const FunctionPointer PROGMEM mainGameLoop[] = {
   stateMenuReboot,
 };
 
-int main() {
-  arduboy.mainNoUSB();
-  return 0;
-}
-
 void setup() {
   arduboy.boot();
   arduboy.audio.begin();
@@ -85,7 +80,7 @@ void loop() {
   drawTiles();
   updateEyes();
   
-  ((FunctionPointer) pgm_read_word(&mainGameLoop[gameState]))();
+  pgm_read_ptr(&mainGameLoop[gameState])();
   checkInputs();
   if (question) drawQuestion();
   if (yesNo) drawYesNo();
