--- a/toyokumono.ino
+++ b/toyokumono.ino
@@ -18,9 +18,9 @@
 
 /*  Local Functions (macros)  */
 
-#define callInitFunc(idx)   ((void (*)(void)) pgm_read_word((uint16_t) &moduleTable[idx].initFunc))()
-#define callUpdateFunc(idx) ((MODE_T (*)(void)) pgm_read_word((uint16_t) &moduleTable[idx].updateFunc))()
-#define callDrawFunc(idx)   ((void (*)(void)) pgm_read_word((uint16_t) &moduleTable[idx].drawFunc))()
+#define callInitFunc(idx) (moduleTable[idx].initFunc)()
+#define callUpdateFunc(idx) (moduleTable[idx].updateFunc)()
+#define callDrawFunc(idx) (moduleTable[idx].drawFunc)()
 
 /*  Local Variables  */
 

