--- a/knightmove.ino
+++ b/knightmove.ino
@@ -6,9 +6,9 @@
 
 /*  Defines  */
 
-#define callInitFunc(idx)   ((void (*)(void)) pgm_read_ptr((uint16_t) &moduleTable[idx]))()
-#define callUpdateFunc(idx) ((MODE_T (*)(void)) pgm_read_ptr((uint16_t) &moduleTable[idx] + 2))()
-#define callDrawFunc(idx)   ((void (*)(void)) pgm_read_ptr((uint16_t) &moduleTable[idx] + 4))()
+#define callInitFunc(idx) (moduleTable[idx].initFunc)()
+#define callUpdateFunc(idx) (moduleTable[idx].updateFunc)()
+#define callDrawFunc(idx) (moduleTable[idx].drawFunc)()
 
 /*  Typedefs  */
 
--- a/title.cpp
+++ b/title.cpp
@@ -29,7 +29,7 @@
 static void drawTitleImage(void);
 static void drawRecord(void);
 static void drawCredit(void);
-static void drawText(const char *p, int lines);
+static void drawText(const char *p, int16_t y);
 
 /*  Local Variables  */
 
--- a/MyArduboy.cpp
+++ b/MyArduboy.cpp
@@ -19,7 +19,7 @@
 
 MyArduboy::MyArduboy(void)
 {
-    pTunes = new ArduboyPlaytune(myAudio.enabled);
+    pTunes = ardugirl_create_playtune(myAudio.enabled);
     pTunes->initChannel(PIN_SPEAKER_1);
 }
 

